tools/oniontracetools/analysis.py

```diff
@@ -351,17 +363,26 @@ def __parse_line(self, line, do_complete):
                         create_dt = datetime.datetime.strptime(time_str, "%Y-%m-%dT%H:%M:%S.%f")
                         break
 
-                if create_dt is not None:
-                    cbt = (now_dt - create_dt).total_seconds()
-
+                if is_cannibalized:
+                    # We can't calculate accurate build or fail times
+                    # for cannibalized circuits, so we just record the counts.
+                    # https://gitlab.torproject.org/tpo/core/tor/-/issues/40570
                     if is_built:
-                        self.circuit['build_time'].setdefault(second, [])
-                        self.circuit['build_time'][second].append(cbt)
-                        self.circuit_summary['circuits_built_total'] += 1
+                        self.circuit_summary['cannibalized_circuits_built_total'] += 1
                     else:
-                        self.circuit['fail_time'].setdefault(second, [])
-                        self.circuit['fail_time'][second].append(cbt)
-                        self.circuit_summary['circuits_failed_total'] += 1
+                        self.circuit_summary['cannibalized_circuits_failed_total'] += 1
+                else:
+                    if create_dt is not None:
+                        cbt = (now_dt - create_dt).total_seconds()
+
+                        if is_built:
+                            self.circuit['build_time'].setdefault(second, [])
+                            self.circuit['build_time'][second].append(cbt)
+                            self.circuit_summary['circuits_built_total'] += 1
+                        else:
+                            self.circuit['fail_time'].setdefault(second, [])
+                            self.circuit['fail_time'][second].append(cbt)
+                            self.circuit_summary['circuits_failed_total'] += 1
```

Comment:
Could we make `self.circuit_summary['circuits_failed_total']` and `self.circuit_summary['cannibalized_circuits_failed_total']` dictionaries that map failure reasons to the number of times they occurred? Something like:
```python
match = re.search("\sREASON=([^\s]+)", line)
reason = match.group(1)
...
self.circuit_summary['circuits_failed_total'].setdefault(reason, 0)
self.circuit_summary['circuits_failed_total'][reason] += 1
```

Reply:
LGTM, thanks!